SpecialStack.java

```diff
@@ -0,0 +1,71 @@
+import java.util.*;
+public class SpecialStack {
+
+    Stack<Integer> stack;
+    Stack<Integer> minStack;
+
+    SpecialStack(){
+        this.stack = new Stack<Integer>();
+        this.minStack = new Stack<Integer>();
+    }
+
+    public boolean isEmpty() 
+    {         
+        return stack.empty();
+    }
+
+    public void push(int x){
+
+        stack.push(x);
+        
+        // to maintain a minStack
+        Stack<Integer> tempStack = new Stack<Integer>();
+        
+        if(minStack.empty()){
+            minStack.push(x);
+            return;
+        }
+        if(minStack.peek() >= x){
+            minStack.push(x);
+        }
+        else{
+            while(!minStack.empty() && minStack.peek() < x ){
+                tempStack.push(minStack.pop());                
+            }
+
+            minStack.push(x);
+
+            while(!tempStack.empty()){
+                int element = (int)tempStack.pop();
+                minStack.push(element);
+            }
+        }       
+    }
+
+    public int pop(){
+        return stack.pop();
+    }
+
+    public int peek(){
+        return stack.peek();
+    }
+
+    public int getMin(){
```

Comment:
Will this work if you pop top few elements out of the stack ?
Will this function still give minimum value. ?
Think about it.

Reply:
I did not run my code against leetcode test cases so ended up missing many edge cases. Made changes to both problems based on inputs from sessions. 

https://github.com/super30admin/Day1/pull/6/commits/4346bce898f79580bd5a988f1c6ceddbc5303551